pygments/lexers/usd.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""The module that parses Pixar's Universal Scene Description file format."""
+
+from pygments import lexer, token
+
+from . import _usd_builtins
+
+__all__ = ["UsdLexer"]
+
+
+_PUNCTUATION = [
+    (r"\(", token.Punctuation),
+    (r"\)", token.Punctuation),
+    (r"\[", token.Punctuation),
+    (r"\]", token.Punctuation),
+    ("{", token.Punctuation),
+    ("}", token.Punctuation),
+]
+
+
+def _keywords(words, type_):
+    """list[tuple[:class:`pygments.lexer.words`, :class:`pygments.token._TokenType`]]."""
+    return [(lexer.words(words, prefix=r"\b", suffix=r"\b"), type_)]
+
+
+_TYPE = r"(\w+(?:\[\])?)"
+_BASE_ATTRIBUTE = r"([\w_]+(?:\:[\w_]+)*)(?:(\.)(timeSamples))?"
+_WHITESPACE = r"([ \t]+)"
+
+
+class UsdLexer(lexer.RegexLexer):
+    """
+    A lexer that parses Pixar's Universal Scene Description file format.
+
+    .. versionadded:: 2.6.0
+    """
+
+    name = "USD"
+    aliases = ["usd", "usda"]
+    filenames = ["*.usd", "*.usda"]
+
+    tokens = {
+        "root": [
+            (
+                r"(custom){_WHITESPACE}(uniform)(\s+){}(\s+){}(\s*)(=)".format(
+                    _TYPE, _BASE_ATTRIBUTE, _WHITESPACE=_WHITESPACE
```

Comment:
What's the difference here between using `WHITESPACE` and `\s+`?

Reply:
When \s was used, \n was getting matched. This isn't allowed by USD. It has to be separated by space or tabs. If there's an alternative that's more expressive, please suggest it and I'll add it

Reply:
ok, I overlooked that difference!